src/node_metadata.cc

```diff
@@ -105,6 +110,48 @@ Metadata::Versions::Versions() {
 #endif
 }
 
+bool hasEnding(std::string const& fullString, std::string const& ending) {
+  if (fullString.length() >= ending.length()) {
+    return (0 == fullString.compare(fullString.length() - ending.length(),
+                                    ending.length(),
+                                    ending));
+  } else {
+    return false;
+  }
+}
+
+std::string Metadata::Versions::GetDependencyVersionFromPackageJson(
+    std::string packageName) {
+
+  // get current path and find parent which has ending 'node'
+  std::filesystem::path cwd = std::filesystem::current_path();
+  while (!hasEnding(cwd.parent_path().string(), "node")) {
+    cwd = cwd.parent_path();
+  }
+  std::ifstream myFile(cwd.parent_path().string()+"\\deps\\" + packageName + "\\src\\package.json");
```

Comment:
Unfortunately, this is not going to work. This file won't exist for the binaries available from nodejs.org.

Reply:
@richardlau How about updating the `update` script for `undici` to statically update a C++ variable with the latest version number?

Reply:
Would this also be an option, read versions and write them in a text file in build step, then read that file to return missing versions? File and node.exe can be in the same directory.

Reply:
Node is compiled into a single binary so you cannot assume any other files are present at run time.

Reply:
You can create a dep-versions.cc and replace the value on tools/deps_updater scripts either through bash or javascript.